pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -149,6 +149,14 @@ func ApplyMonitoringConsole(ctx context.Context, client splcommon.ControllerClie
 	if cr.Status.Phase == enterpriseApi.PhaseReady {
 		finalResult := handleAppFrameworkActivity(ctx, client, cr, &cr.Status.AppContext, &cr.Spec.AppFrameworkConfig)
 		result = *finalResult
+
+		// TODO: Fix the Change Annotation logic/ find an alternative (eg. state machine); right now the search head deployer
```

Comment:
Is there any reason why the replicas don't come up properly?

Reply:
While running the tests the deployer was going down and some times (not every time) the replica-2 went down and went into a restart loop. This got fixed after rerunning the tests though. There is still one test case that fails in the 1st commit of the PR which can pass eventually (it checks if the LM is configured properly and because replica - 2 goes down this test fails but will pass if we wait for sometime).

Reply:
Is there a reason why the pods are going down? Ideally we should be taking care of those in this MR or create a task to take care of it later.

Reply:
From what I gathered from the test cases it is happening because of the change annotation which is why I have removed the call for that in the final commit ( the pods are not going down now). We had earlier thought that the pods should not restart with a change in the annotations but it is happening for the search head

Reply:
@vivekr-splunk I believe we are appending the annotations from CR to the statefulSet [here](https://github.com/splunk/splunk-operator/blob/6f2e605134adf7a7d07310e1f1651797d1091121/pkg/splunk/enterprise/configuration.go#L709). Since we identify the differences [here](https://github.com/splunk/splunk-operator/blob/6f2e605134adf7a7d07310e1f1651797d1091121/pkg/splunk/controller/util.go#L61) we end up doing a restart of all the pods of the SS. The time taken to restart the deployer/Search heads seems to be the issue here - it is failing the tests.